File: server/src/routers/priority.ts
import { PriorityRoute } from '@/enums';
import { usePriority } from '@/services';
import { Hono } from 'hono';

const priority = new Hono();
const service = usePriority();

priority.get(PriorityRoute.RETRIEVE, async (c) => c.json(await service.list()));

export { priority };

File: server/internal/services/priority.go
package services

import (
	"sukejuuru/internal/models"
	"sukejuuru/internal/utils"

	"github.com/jmoiron/sqlx"
)

type priorityService struct {
	db *sqlx.DB
}

func UsePriority() *priorityService {
	return &priorityService{db: utils.UsePostgres()}
}

func (service priorityService) FindAll() ([]models.Priority, error) {
	var priorities []models.Priority
	query := "SELECT id, display, description, created_at, color FROM priority"
	if err := service.db.Select(&priorities, query); err != nil {
		return nil, err
	}
	return priorities, nil
}
